Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionDynamicScriptsCocoa.mm

```diff
@@ -196,6 +193,89 @@ WebExtensionScriptInjectionResultParameters toInjectionResultParameters(id resul
     return parameters;
 }
 
+WebExtensionRegisteredScript::WebExtensionRegisteredScript(WebExtensionContext& extensionContext, const WebExtensionRegisteredScriptParameters& parameters)
+    : m_extensionContext(extensionContext)
+    , m_parameters(parameters)
+{
+
+}
+
+void WebExtensionRegisteredScript::updateParameters(WebExtensionRegisteredScriptParameters& parameters)
+{
+    m_parameters = parameters;
+}
+
+void WebExtensionRegisteredScript::merge(WebExtensionRegisteredScriptParameters& parameters)
+{
+    if (!parameters.css && m_parameters.css)
+        parameters.css = m_parameters.css.value();
+
+    if (!parameters.js && m_parameters.js)
+        parameters.js = m_parameters.js.value();
+
+    if (!parameters.injectionTime)
+        parameters.injectionTime = m_parameters.injectionTime.value();
+
+    if (!parameters.excludeMatchPatterns && m_parameters.excludeMatchPatterns)
+        parameters.excludeMatchPatterns = m_parameters.excludeMatchPatterns.value();
+
+    if (!parameters.matchPatterns)
+        parameters.matchPatterns = m_parameters.matchPatterns.value();
+
+    if (!parameters.allFrames)
+        parameters.allFrames = m_parameters.allFrames.value();
+
+    if (!parameters.persistent)
+        parameters.persistent = m_parameters.persistent.value();
+
+    if (!parameters.forMainWorld)
+        parameters.forMainWorld = m_parameters.forMainWorld.value();
+}
+
+void WebExtensionRegisteredScript::addUserScript(String identifier, API::UserScript& userScript)
+{
+    if (m_userScriptsMap.contains(identifier))
+        m_userScriptsMap.get(identifier).append(userScript);
+    else
+        m_userScriptsMap.add(identifier, UserScriptVector { userScript });
```

Comment:
This can be done with `ensure()`.

```
auto& userScripts = m_userScriptsMap.ensure(identifier, [&] {
    return UserScriptVector { };
}).iterator->value;
userScripts.append(userScript);
```